src/libs/history.c

```diff
@@ -169,6 +175,67 @@ static GtkWidget *_lib_history_create_button(dt_lib_module_t *self, int num, con
   return widget;
 }
 
+static GList *_duplicate_history(GList *hist)
+{
+  GList *result = NULL;
+
+  GList *h = g_list_first(hist);
+  while(h)
+  {
+    const dt_dev_history_item_t *old = (dt_dev_history_item_t *)(h->data);
+
+    dt_dev_history_item_t *new = (dt_dev_history_item_t *)g_malloc(sizeof(dt_dev_history_item_t));
+
+    new->module = old->module;
+    new->enabled = old->enabled;
+    new->multi_priority = old->multi_priority;
+    snprintf(new->multi_name, sizeof(old->multi_name), "%s", old->multi_name);
+    new->focus_hash = old->focus_hash;
+
```

Comment:
Why not one memcpy? That way you won't forget fields if anything new gets added in the future.